prowler/lib/check/models.py

```diff
@@ -536,26 +536,34 @@ def __init__(self, metadata: Dict, resource: Any) -> None:
 
 
 @dataclass
-class Check_Report_Microsoft365(Check_Report):
+class CheckReportMicrosoft365(Check_Report):
     """Contains the Microsoft365 Check's finding information."""
 
     resource_name: str
     resource_id: str
     location: str
 
-    def __init__(self, metadata: Dict, resource: Any) -> None:
+    def __init__(
+        self,
+        metadata: Dict,
+        resource: Any,
+        resource_name: str = "",
+        resource_id: str = "",
```

Comment:
I think is highly important not to set default values as empty strings for this two fields to enforce that are always passed with the expected values.